src/main/java/org/broadinstitute/hellbender/tools/gvs/common/WeightedSplitIntervals.java

```diff
@@ -122,24 +121,29 @@ public void onTraversalStart() {
             lastContig = wi.getContig();
 
             // if the interval fits completely, just add it
-            if (cumulativeWeight + wi.getWeight() <= targetWeightPerScatter ) {
+            if (cumulativeWeight + wi.getWeight() <= targetWeightPerScatter) {
                 cumulativeWeight += wi.getWeight();
                 currentList.add(wi);
 
             // if it would push us over the edge
             } else {
-                // add a piece of it
+                // try to add a piece of it
                 float remainingSpace = targetWeightPerScatter - cumulativeWeight;
 
                 // how many bases can we take?
-                int basesToTake = (int) Math.floor(remainingSpace /  wi.getWeightPerBase());
-
-                // split and add the first part into this list
-                WeightedInterval[] pair = wi.split(basesToTake);
-                currentList.add(pair[0]);
-
-                // push the remainder back onto the iterator
-                iter.pushback(pair[1]);
+                int basesToTake = (int) Math.floor(remainingSpace / wi.getWeightPerBase());
+
+                if (basesToTake == 0) {
+                    // We can't add any more bases to the current interval list so put this interval back.
+                    iter.pushback(wi);
```

Comment:
Fixes the Charlie crashing issue and the resulting interval list files are identical to those produced by the `86f1116` version of the gatk jar used for beta

Reply:
looks great